### DISCUSSION TITLE
Add debounce modifier for x-intersect

### DISCUSSION THREAD

**Discussion by acalvino4** [alpinejs/alpine#3703] [3. Feature ideas]
The title says it all.

If you are open to this I can give a stab at a PR.

**Comment by ekwoka** [comment]
:thinking: What does this solve?

How would it intersect multiple times in a short amount of time?

**Comment by acalvino4** [comment]
Good question, I can see how in 99+% of (maybe all?) use cases this would be irrelevant.

I am trying to achieve the shrinking header effect as shown here, but using intersection observer instead of onscroll events: https://www.w3schools.com/howto/howto_js_shrink_header_scroll.asp

I've got it working, but there's a little jank bc as soon as you scroll, the header shrinks, causing you to be at the top of the page again, which then makes the header grow, etc. This is where I thought debounce might help.

...

But upon further thinking I realized debounce wouldn't really solve this issue; I instead I solved it with margin like this:

```html
<body x-data="{top: true}" class="flex flex-col min-h-screen">
    {# An invisible pixel that fires events when scroll position reaches/leaves the top #}
    <div x-intersect:enter="top = true" x-intersect:leave.margin.96="top = false" class="absolute h-px invisible"></div>
```

With this code in place I can apply classes anywhere based on whether I'm at the top of the page, e.g.: `:class="top ? 'h-32' : 'h-24'"`
